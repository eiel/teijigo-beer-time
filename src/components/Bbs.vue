<template>
  <div class="bbs">
    <h1>一言掲示板</h1>
    <div class="message-form">
      <label for="messageInput">メッセージ</label>
      <input type="text" id="messageInput" maxlength="20px" v-model="message">
      <button type="button" @click="sendMessage">送信</button>
    </div>
    <div>
      <ul class="message-list">
        <li v-for="(item, key) in messageList" v-bind:key="key">
          {{item.message}}
        </li>
      </ul>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import database from '../firebase-config.ts';


interface Message {
  message: string,
  sortKey: number
}

interface BbsData {
  messageList: Message[],
  message: string
}

export default Vue.extend({
  name: 'bbs',
  data() :BbsData {
    return {
      messageList: [],
      message: '',
    };
  },
  created() {
    this.listen();
  },
  methods: {
    listen() {
      database.ref('messages/').on('value', (snapshot) => {
        if (snapshot) {
          this.messageList = snapshot.val();
        }
      });
    },
    sendMessage() {
      if (!this.message) return;

      database.ref('messages/').push({
        message: this.message,
        sortKey: -(new Date()),
      });

      this.message = '';
    },
  },
});
</script>

<style scoped>
  .message-form {
    display: flex;
    justify-content: center;
  }

  .message-form label {
    width: 5em;
    font-weight: bold;
    margin-right: 10px;
  }
  .message-form input {
    padding: 5px;
  }

  .message-form button {
    background-color: gray;
    width: 50px;
    color: white;
    padding: 5px;
    margin-right: 10px;
    margin-left: 10px;
  }

  .message-list {
    padding: 0;
    margin-left: auto;
    margin-right: auto;
    list-style: none;
    text-align: left;
    max-width: 90%;
  }

  .message-list li {
    background-color: white;
    padding: 5px 10px;
    margin-bottom: 5px;
  }
</style>
