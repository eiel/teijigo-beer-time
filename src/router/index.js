import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/Top';
import Disco from '@/components/Disco';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
  ],
});
