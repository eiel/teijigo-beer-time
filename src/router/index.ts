import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/Top.vue';
import Disco from '@/components/Disco.vue';
import NotFound from '@/components/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
});
