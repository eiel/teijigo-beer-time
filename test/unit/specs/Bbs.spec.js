import Vue from 'vue';
import Bbs from '@/components/pages/Bbs';
import router from '@/router/index';

describe('Bbs.vue', () => {
  const Constructor = Vue.extend(Bbs);
  const vm = new Constructor({ router }).$mount();

  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot()
  });
});