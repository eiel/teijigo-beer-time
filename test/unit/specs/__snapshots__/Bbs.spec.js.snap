// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bbs.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menu-item"
      >
        <a
          class="router-link-active"
          href="/"
        >
          
          トップ
        
        </a>
      </li>
       
      <li
        class="menu-item"
      >
        <a
          class=""
          href="/disco"
        >
          
          DISCOGRAPHY
        
        </a>
      </li>
       
      <li
        class="menu-item"
      >
        <a
          class=""
          href="/member"
        >
          
          メンバー紹介
        
        </a>
      </li>
       
      <li
        class="menu-item"
      >
        <a
          class=""
          href="/bbs"
        >
          
          一言掲示板
        
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    id="main-container"
  >
    <a
      class="router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="bbs"
    >
      <!---->
       
      <p>
        
      足跡や、あいさつ、キリ番報告、等に使用していただいて構いません。
      
        <br />
        
      誹謗中傷、個人情報などの書き込みはご遠慮ください。
    
      </p>
       
      <div
        class="message-form"
      >
        <div
          class="message-form-group"
        >
          <label
            for="nameInput"
          >
            HN(ハンドルネーム)
          </label>
           
          <input
            id="nameInput"
            maxlength="20"
            type="text"
          />
        </div>
         
        <div
          class="message-form-group"
        >
          <label
            for="messageInput"
          >
            メッセージ
          </label>
           
          <input
            id="messageInput"
            maxlength="100"
            type="text"
          />
        </div>
         
        <button
          type="button"
        >
          送信
        </button>
      </div>
       
      <div>
        <ul
          class="message-list"
        />
      </div>
    </div>
     
    <footer>
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
